tests/test_registrable.py

```diff
@@ -160,3 +169,26 @@ class Test(Registrable):
     # Function returns class name if no entry in registry was found
     assert A.name_for_object_class(c) == "C"
     assert A.name_for_object_class(test) == "Test"
+
+
+def test_auto_classes():
+    class Base(Registrable):
+        pass
+
+    T = TypeVar("T", bound=RegistrableProtocol)
+
+    class Auto(Registrable[T]):
```

Comment:
Don't call this `Auto`. First, there is already a class `pie_core.Auto`. Second, "Auto" is just one "usecase". In this test, it should be named as what it is: kind of a "not registered registrable"...?